data/multiplayer/scenarios/2p_Isle_of_Mists.cfg

```diff
@@ -0,0 +1,298 @@
+        team_name="human"
+        user_team_name= _ "teamname^Defenders"
+        controller="human"
+        save_id="Player2"
+        gold=100
+        income=0
+        gold_lock=yes
+        {FLAG_VARIANT long}
+        income_lock=yes
+        controller_lock=no
+        team_lock=yes
+        faction_lock=no
+        leader_lock=no
+        color_lock=no
+        fog=yes
+        share_vision=all
+        [variables]
+            dont_make_me_quick=yes
+        [/variables]
+    [/side]
+
+    [side]
+        side=5
+        color=black
+        team_name="human"
+        user_team_name= _ "teamname^Defenders"
+        allow_player=no
+        hidden=yes
+        disallow_observers=yes
+        {FLAG_VARIANT long}
+        no_leader=yes
+        controller="ai"
+    [/side]
+
+    [lua]
+        code = << wesnoth.dofile( "multiplayer/scenarios/2p_Isle_of_Mists.lua" ) >>
+    [/lua]
+
+    [event]
+        name=prestart
+        [store_unit]
+            variable=leader
+            [filter]
+                side=3,4
+            [/filter]
+            kill=no
+        [/store_unit]
+        [if]
+            [variable]
+                name=leader.length
+                less_than=2
+            [/variable]
+            [then]
+                {VARIABLE lose_condition_string ( _ "Death of your leader")}
+                {VARIABLE notes ( _ "Since your team has only one leader, the enemy waves will not be at full strength.")}
+            [/then]
+            [else]
+                {VARIABLE lose_condition_string ( _ "Death of both of your team’s leaders")}
+                {VARIABLE notes ( _ "Since your team has two leaders, the enemy waves will be at full strength.")}
+                # each village will now support an upkeep of 2 instead of 1
+                # this makes the scenario somewhat passable now.
+                [modify_side]
+                    side=3,4
+                    village_support=2
+                [/modify_side]
+            [/else]
+        [/if]
+        {CLEAR_VARIABLE leader}
+        [music]
+            name=wanderer.ogg
+            ms_before=2000
+        [/music]
+        [music]
+            name=battle.ogg
+            ms_before=2000
+            append=yes
+        [/music]
+        [music]
+            name=breaking_the_chains.ogg
+            play_once=yes
+            immediate=yes
+        [/music]
+
+        [objectives]
+            [objective]
+                description= _ "Survive and defeat all enemy waves"
+                condition=win
+            [/objective]
+            [objective]
+                description=$lose_condition_string
+                condition=lose
+            [/objective]
+
+            [note]
+                description=$notes
+            [/note]
+        [/objectives]
+
+        {CLEAR_VARIABLE lose_condition_string}
+        {CLEAR_VARIABLE notes}
+
+        [label]
+            x,y = 16,9
+            text = _"Healing Keep"
+        [/label]
```

Comment:
It would be good to remove this label when the Elvish Lady dies.

Reply:
Done